### PR TITLE
Hunter Casting Times

### PR DESCRIPTION
**Pull request by dagochen**
Current situation (issues which are fixed with this PR):
Multishot casting time is not improved by rapid fire, quick shots or anything else.
Aimed Shot casting time is improved by quiver.
Aimed Shot casting time is decreased by 0,5 seconds for no reason.

What does this PR do?
Aimed shot must not profit from quiver (any hunter guide, any youtube video...) and the casting time is set to 3 seconds. It improves correctly by rapid fire, same as the multi shot.

Proof:
https://youtu.be/ACM3yXuz0Hw?t=1m30s
Aimed shot needs 3 seconds with quiver equipped

Spelleffects: 
Quiver: 14829 - Apply Aura: Haste - Ranged (Ammo Only) 
Rapid Fire: 3045 - Apply Aura: Haste - Ranged
Quick Shots: 6150 - Apply Aura: Haste - Ranged

Any hunter guide who says that Ashjre'tul is best, because of the slow weapon speed. If Aimed shot would get bonus from quiver, all ranged weapons with higher attack rate would be better.

### CODE REVIEW

File: src/game/SpellMgr.cpp [hunk 1]
```diff
@@ -120,23 +122,29 @@ uint32 GetSpellCastTime(SpellEntry const* spellInfo, Spell const* spell)
...
-        {
-            if (spell->IsRangedSpell() && !spell->IsAutoRepeat())
-                castTime = int32(castTime * spell->GetCaster()->m_modAttackSpeedPct[RANGED_ATTACK]);
-        }
     }
-
-    if (spellInfo->HasAttribute(SPELL_ATTR_RANGED) && (!spell || !spell->IsAutoRepeat()))
-        castTime += 500;
-
+
```
Comment (Phatcat) [inline src/game/SpellMgr.cpp]: Trailing whitespaces

File: src/game/SpellMgr.cpp [hunk 2]
```diff
@@ -26,6 +26,8 @@
 #include "Spell.h"
 #include "Unit.h"
 #include "World.h"
+#include "SpellAuras.h"
+
```
Comment (Phatcat) [inline src/game/SpellMgr.cpp]: There's no reason for this empty line when you have one just on the next line =)

File: src/game/SpellMgr.cpp [hunk 3]
```diff
@@ -120,23 +122,29 @@ uint32 GetSpellCastTime(SpellEntry const* spellInfo, Spell const* spell)
 
     int32 castTime = spellCastTimeEntry->CastTime;
 
+    // Multishot only (alternative set CastingTimeIndex = 3)
+    if (spellInfo->HasAttribute(SPELL_ATTR_RANGED) && (!spell || !spell->IsAutoRepeat()) && spellInfo->SpellFamilyFlags & 0x0000000000001000) 
+        castTime = 500;
+
```
Comment (Phatcat) [inline src/game/SpellMgr.cpp]: Trailing whitespaces

**Comment by dagochen** [comment]
#freeTrailingWhitespaces :P

I ll take care of them tonight ;)

**Comment by killerwife** [comment]
@dagochen there is something in trinity, you would very much like SPELL_ATTR2_NOT_RESET_AUTO_ACTIONS

**Comment by dagochen** [comment]
Haha ;)

**Comment by dagochen** [comment]
Shall I fix this up for merge?

**Comment by Phatcat** [comment]
Well, someone somewhat familiar with the spell system and hunter behaviour would need to review and test it before merge and if there are corrections you may end up having to fix up again.

Your call =)

**Comment by dagochen** [comment]
Done. This should be fine at least ;)

**Comment by dagochen** [comment]
@cyberium @cala Someone can have a look at this?

**Pull request closed without merging**